### DISCUSSION TITLE
Is there a guide for getting started, using prebuilt libraries?

### DISCUSSION THREAD

**Discussion by tastewar** [ned14/llfio#145] [Q&A] [answered]
My company's portfolio of projects generally uses conan for managing libraries, so I started down the path of creating a conan recipe for llfio. I saw that there are existing conan recipes for outcome and quickcpplib, and I was able to create one for ntkernel-error-category, but when I started looking at the CMakeLists for llfio, I realized I was way out of my depth. So I thought that perhaps I could instead get started by using the pre-built libraries. I downloaded the linux pre-build files for the 20240823 release. I copied the include directory into my project, and I #included the top level llfio.hpp file after #define'ing LLFIO_STATIC_LINK but I get an error:

[Build:x64d] /home/tastewar/xp/raflibii/source/../llfio/llfio/v2.0/config.hpp:109:10: fatal error: quickcpplib/cpp_feature.h: No such file or directory
[Build:x64d] 109 | #include "quickcpplib/cpp_feature.h"

The directory I copied in (from the pre-built binaries download) has quickcpplib and outcome directories. It looks like the quickcpplib folder has its own include folder, and another layer deep, there is a quickcpplib folder, and in there is the desired cpp_feature.h file.

Am I supposed to be re-arranging things after downloading and extracting? If so, is this documented somewhere? Or did I miss a step in setting this up? Is there some other pre-processor symbol I need to define?

Thanks for any help you can offer!
--Tom

**Comment by ned14** [comment, marked as answer]
For the prebuilt binaries distro, you need to tell your compiler about the individual `include` paths, then it will find all the headers. For example:

- `include`
- `include/quickcpplib/include`
- `include/outcome/include`

If you wish to rearrange things, you can do so. Up to you. The reason it is like it currently is is more historical accident than anything intentional. Back when the prebuilt binaries was set up, cmake install hadn't been implemented yet.

For most people it's probably easier to use the cmake and go from there.

There is a live vcpkg for LLFIO. There isn't one for Conan. Contributions welcome, studying the vcpkg edition would probably be useful.

Note that LLFIO can be a header only library, so once the include paths are set up, it should "just work".

**Reply by tastewar** [reply]
Thanks for the help!
